File: src/main/java/engine/core/util/DeltaTimeCalculator.java
package engine.core.util;

public class DeltaTimeCalculator {

    private long lastTime;

    public DeltaTimeCalculator() {
        lastTime = System.nanoTime();
    }

    public float calcDeltaTime() {
        long time = System.nanoTime();
        float deltaTime = ((time - lastTime) / 1000000f);
        lastTime = time;

        return deltaTime;
    }
}

File: src/main/java/engine/util/DeltaTimeCalculator.kt
package engine.util

class DeltaTimeCalculator {
    private var lastTime = System.nanoTime()

    fun calcDeltaTime(): Float {
        val time = System.nanoTime()
        val deltaTime = (time - lastTime) / 1000000f
        lastTime = time

        return deltaTime
    }
}
